network/src/peer_manager/mod.rs

```diff
@@ -100,7 +104,7 @@ impl PeerManagerRequestSender {
         let (oneshot_tx, oneshot_rx) = oneshot::channel();
         let request = PeerManagerRequest::DialPeer(peer_id, addr, oneshot_tx);
         self.inner
-            .push((peer_id, ProtocolId::default()), request)
+            .push((peer_id, ProtocolId::from_static(b"DialPeer")), request)
```

Comment:
I almost wonder if it might be worth separating our Dial/DisconnectPeer to a separate LIFO-1 channel like we've done for New/LostPeer. Something to think about for another, later diff.